langkit/ocaml_api.py

```diff
@@ -0,0 +1,444 @@
+            lambda x: not x.is_overriding and not x.abstract and not x.null
+        )
+
+    def get_properties(self, tpe):
+        """
+        Return the list of all properties that should be exposed to the user.
+
+        :rtype: list[ct.BaseField]
+        """
+
+        return [
+            field
+            for field in tpe.fields_with_accessors()
+            if field.is_property
+        ]
+
+    def array_wrapper(self, array_type):
+        return (ct.T.entity.array
+                if array_type.element_type.is_entity_type else
+                array_type).api_name.camel
+
+    def struct_name(self, type):
+        """
+        Returns the ocaml module containing the low-level structure for the
+        given type.
+
+        :param ct.CompiledType type: Type we want to get the module name.
+        :rtype: str
+        """
+
+        return dispatch_on_type(type, [
+            (ct.EnumType, lambda t: self.module_name(t)),
+            (T.Token, lambda t: self.module_name(t)),
+            (T.Symbol, lambda t: self.module_name(t)),
+            (ct.ASTNodeType, lambda t: self.module_name(t.entity)),
+            (ct.EntityType, lambda _: 'EntityStruct'),
+            (T.AnalysisUnit, lambda t: '{}Struct'.format(self.module_name(t))),
+            (ct.ArrayType, lambda t: '{}Struct'.format(self.array_wrapper(t))),
+            (ct.StructType, lambda t: '{}Struct'.format(self.module_name(t))),
+            (T.BigInt, lambda t: self.module_name(t)),
+        ])
+
+    def node_name(self, type):
+        """
+        Returns the OCaml name for an ast node type or an entity type.
+
+        :param ct.EntityType | ct.ASTNodeType type: Type we want to get the
+            name.
+        :rtype: str
+        """
+        return dispatch_on_type(type, [
+            (ct.ASTNodeType, lambda t: self.node_name(t.entity)),
+            (ct.EntityType, lambda t: t.astnode.kwless_raw_name.camel),
+        ])
+
+    def module_name(self, type):
+        """
+        Returns the ocaml module containing the definition for the given type.
+
+        :param ct.CompiledType type: Type we want to get the module name.
+        :rtype: str
+        """
+
+        return dispatch_on_type(type, [
+            (ct.EnumType, lambda enum: enum.api_name.camel),
+            (T.Token, lambda _: 'Token'),
+            (T.Symbol, lambda _: 'Symbol'),
+            (T.Character, lambda _: 'Character'),
+            (ct.ASTNodeType, lambda t: self.module_name(t.entity)),
+            (ct.EntityType, lambda t: '{}Type'.format(self.node_name(t))),
+            (T.AnalysisUnit, lambda t: t.api_name.camel),
+            (ct.ArrayType, lambda t: t.api_name.camel),
+            (ct.StructType, lambda t: t.api_name.camel),
+            (T.BigInt, lambda t: 'BigInteger'),
+        ])
+
+    def wrap_requires_context(self, type):
+        """
+        Returns true if the given type need the context to be wrapped in an
+        ocaml value.
```

Comment:
Casing of "OCaml" is inconsistent. Pick one ;)

Reply:
I vote for OCaml as soon as we designate the language :-) (not the interpreter program)